### ISSUE TITLE
Cannot access without adding clusterid

### ISSUE THREAD

**Issue by gutsola** [submariner-io/submariner#3167]
**What happened**:
I set up three clusters:  broker, c1, and c2, all of them has only master node. After that, i use `subctl join --kubeconfig ./config/c2.yaml broker-info.subm --clusterid c2 --natt=false` make c2 a member .
![image](https://github.com/user-attachments/assets/689c8734-9c28-49c8-891d-bebb90d3def7)

In c1, I use `subctl export service --namespace sample whereami-cs`  exported the service whereami-cs which in the namespace sample. In c2, I couldn't access it use `curl whereami-cs.sample.svc.clusterset.local`, but when i use `curl c1.whereami-cs.sample.svc.clusterset.local`, it worked.

Client pod info:
kubectl run client --image=cr7258/nettool:v1
all test curl command exec in pod `client`.

info:
use c1.whereami-cs.sample.svc.clusterset.local
![image](https://github.com/user-attachments/assets/b397d278-dc0f-4ccd-91f5-70bd400de9c6)

use whereami-cs.sample.svc.clusterset.local
![image](https://github.com/user-attachments/assets/3f476ae9-2927-4eb2-897b-f34346f162cf)

test-submariner.yaml
`
apiVersion: apps/v1
kind: Deployment
metadata:
  name: whereami
  namespace: sample
spec:
  replicas: 1
  selector:
    matchLabels:
      app: whereami
  template:
    metadata:
      labels:
        app: whereami
    spec:
      containers:
      - name: whereami
        image: cr7258/whereami:v1
        imagePullPolicy: IfNotPresent
        ports:
        - containerPort: 80
        env:
        - name: NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: POD_IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
---
apiVersion: v1
kind: Service
metadata:
  name: whereami-cs
  namespace: sample
spec:
  selector:
    app: whereami
  ports:

`

**What you expected to happen**:
use  `curl whereami-cs.sample.svc.clusterset.local` successful.

**How to reproduce it (as minimally and precisely as possible)**:
Install broker first(will as broker role)
subctl join --kubeconfig ./config/c1.yaml broker-info.subm --clusterid c1 --natt=false
subctl join --kubeconfig ./config/c2.yaml broker-info.subm --clusterid c2 --natt=false

**Anything else we need to know?**:
what should i do to access "whereami-cs.sample.svc.clusterset.local" successfully in cluster c2.
when use nslookup:
![image](https://github.com/user-attachments/assets/777eb985-c867-4857-8bcc-5c6c4d17e68f)

complement:
i change test client from cr7258/nettool:v1 to alpine3.19 , use nslookup successful, but curl failed.
like follow:
![image](https://github.com/user-attachments/assets/8dd36f99-2382-4c63-8bb2-aef335369f88)

I'm new to k8s and submariner, need some help. 

**Environment**:
- k3s: v1.27.2+k3s1
- subctl: v0.13.0
- docker: 20.10.9

**Comment by yboaron** [comment]
Thanks for trying Submariner @trumpzhang ,

Please share the output of [1] from both clusters


[1] 
`subctl show all

subctl diagnose all
`

**Comment by gutsola** [comment]
> Thanks for trying Submariner @trumpzhang ,
> 
> Please share the output of [1] from both clusters
> 
> [1] `subctl show all
> 
> subctl diagnose all `

thanks your reply. Query result like below:
- broker

subctl show all
![image](https://github.com/user-attachments/assets/c000e9ae-d9a1-444b-803f-e7af67bcc671)

subctl diagnose all
![image](https://github.com/user-attachments/assets/9b7b02b1-fcc2-45e6-a7df-090cb0f736af)

- c1

subctl --kubeconfig ./config/c1.yaml show all
![image](https://github.com/user-attachments/assets/975784d7-732e-4c65-a900-9ac55f7b765b)

subctl --kubeconfig ./config/c1.yaml diagnose all
![image](https://github.com/user-attachments/assets/9ec2c3fb-5737-459c-b5eb-881b7bc8c370)

- c2

subctl --kubeconfig ./config/c2.yaml show all
![image](https://github.com/user-attachments/assets/87e73c39-70c8-446a-a54d-e20e8222c51a)

subctl --kubeconfig ./config/c2.yaml diagnose all
![image](https://github.com/user-attachments/assets/d759759f-79ad-4816-b22f-3394b1617e79)

**Comment by trumpzhang** [comment]
By the way, I change test client from cr7258/nettool:v1 to alpine:3.19.
In pod alpine:
``` bash
[root@192 karmada]# kubectl  exec -it alpine-deployment-85cd8fcdcb-j688t sh
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl kubectl exec [POD] -- [COMMAND] instead.
/ #
/ # nslookup whereami-cs.sample.svc.clusterset.local
Server:         10.45.0.10
Address:        10.45.0.10:53

Name:   whereami-cs.sample.svc.clusterset.local
Address: 10.47.178.75


/ # curl 10.47.178.75
{
  "dest_ip": "10.46.0.15",
  "dest_port": 80,
  "http_headers": {
    "Accept": [
      "*/*"
    ],
    "User-Agent": [
      "curl/8.9.1"
    ]
  },
  "namespace": "sample",
  "node_name": "192.168.170.81-master",
  "pod_name": "whereami-58b45d7dbf-csxhc",
  "source_ip": "10.44.0.48",
  "source_port": 36228,
  "timestamp": "2024-09-24T10:12:12Z"
}/ #
/ # curl whereami-cs.sample.svc.clusterset.local
curl: (52) Empty reply from server
/ # curl whereami-cs.sample.svc.clusterset.local -v
* Host whereami-cs.sample.svc.clusterset.local:80 was resolved.
* IPv6: (none)
* IPv4: 198.18.0.68
*   Trying 198.18.0.68:80...
* Connected to whereami-cs.sample.svc.clusterset.local (198.18.0.68) port 80
> GET / HTTP/1.1
> Host: whereami-cs.sample.svc.clusterset.local
> User-Agent: curl/8.9.1
> Accept: */*
>
* Request completely sent off
* Empty reply from server
* shutting down connection #0
curl: (52) Empty reply from server


# when use c1.whereami-cs.sample.svc.clusterset.local
/ # nslookup c1.whereami-cs.sample.svc.clusterset.local
Server:         10.45.0.10
Address:        10.45.0.10:53

Name:   c1.whereami-cs.sample.svc.clusterset.local
Address: 10.47.178.75


/ #
/ # curl c1.whereami-cs.sample.svc.clusterset.local -v
* Host c1.whereami-cs.sample.svc.clusterset.local:80 was resolved.
* IPv6: (none)
* IPv4: 10.47.178.75
*   Trying 10.47.178.75:80...
* Connected to c1.whereami-cs.sample.svc.clusterset.local (10.47.178.75) port 80
> GET / HTTP/1.1
> Host: c1.whereami-cs.sample.svc.clusterset.local
> User-Agent: curl/8.9.1
> Accept: */*
>
* Request completely sent off
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Tue, 24 Sep 2024 10:14:14 GMT
< Content-Length: 353
<
{
  "dest_ip": "10.46.0.15",
  "dest_port": 80,
  "http_headers": {
    "Accept": [
      "*/*"
    ],
    "User-Agent": [
      "curl/8.9.1"
    ]
  },
  "namespace": "sample",
  "node_name": "192.168.170.81-master",
  "pod_name": "whereami-58b45d7dbf-csxhc",
  "source_ip": "10.44.0.48",
  "source_port": 37314,
  "timestamp": "2024-09-24T10:14:14Z"
* Connection #0 to host c1.whereami-cs.sample.svc.clusterset.local left intact

```

As above show, access cluster c1 svc ip 10.47.178.75 in c2 is ok. But access svc domain is failed.  nslookup whereami-cs.sample.svc.clusterset.local show that Address is 10.47.178.75, but when i curl -v, it shows:
/ # curl whereami-cs.sample.svc.clusterset.local -v
* Host whereami-cs.sample.svc.clusterset.local:80 was resolved.
* IPv6: (none)
* IPv4: 198.18.0.68
why the "IPv4" is not "10.47.178.75" but "198.18.0.68". Maybe my Virtual Host has some err config? I use centos7.x.

**Comment by trumpzhang** [comment]
I updated c2 Host Node 's /etc/resolv.conf. 
Added submariner-lighthouse-coredns ClusterIP "10.45.181.68" to Host Node 's /etc/resolv.conf


```bash
[root@192 submariner]# kubectl get svc -A|grep 10.45.181.68
submariner-operator   submariner-lighthouse-coredns           ClusterIP      10.45.181.68    <none>           53/UDP                       168m

[root@192 submariner]# cat /etc/resolv.conf
# submariner-lighthouse-coredns 
nameserver 10.45.181.68
nameserver 8.8.8.8
nameserver 114.114.114.114

```

Then, try curl in Host, it worked. Like :
```bash
[root@192 submariner]# curl whereami-cs.sample.svc.clusterset.local
{
  "dest_ip": "10.46.0.15",
  "dest_port": 80,
  "http_headers": {
    "Accept": [
      "*/*"
    ],
    "User-Agent": [
      "curl/7.29.0"
    ]
  },
  "namespace": "sample",
  "node_name": "192.168.170.81-master",
  "pod_name": "whereami-58b45d7dbf-csxhc",
  "source_ip": "10.44.0.0",
  "source_port": 32892,
  "timestamp": "2024-09-24T10:51:17Z"
}
```

In test pod , use try, it still not worked.
```bash
/ # curl whereami-cs.sample.svc.clusterset.local
curl: (52) Empty reply from server
/ #
/ # curl whereami-cs.sample.svc.clusterset.local -v
* Host whereami-cs.sample.svc.clusterset.local:80 was resolved.
* IPv6: (none)
* IPv4: 198.18.0.68
*   Trying 198.18.0.68:80...
* Connected to whereami-cs.sample.svc.clusterset.local (198.18.0.68) port 80
> GET / HTTP/1.1
> Host: whereami-cs.sample.svc.clusterset.local
> User-Agent: curl/8.9.1
> Accept: */*
>
* Request completely sent off
* Empty reply from server
* shutting down connection #0
curl: (52) Empty reply from server
/ #
/ # cat /etc/resolv.conf
nameserver 10.45.0.10
search default.svc.cluster.local svc.cluster.local cluster.local localdomain
options ndots:5
/ #

```

**Comment by yboaron** [comment]
First of all, seems that you are using an older version (0.13) of Submariner, please reinstall Submariner using latest subctl (0.18), check [1] for more details. 

Post deployment, you can use subctl  [automatic verification](https://submariner.io/getting-started/quickstart/kind/#verify-automatically-with-subctl) for testing inter-cluster service discovery functionality.

[1] 
https://submariner.io/operations/deployment/subctl/#installation

**Comment by gutsola** [comment]
@yboaron 
Thanks for your reply. I find the problem and fix it up. 
A few days ago, I installed app named "Clash Verge" which is a proxy software. For convenience, I turned on its Tun mode, which caused requests from pods in my virtual machine to not be resolved correctly. Because I found that the curl public domain name in the pod was unresponsive, but the curl public domain name on the host was good. When I turned off Tun mode, everything worked normally.
